crates/anvil/src/eth/otterscan/api.rs

```diff
@@ -1,19 +1,66 @@
-use super::types::{
-    OtsBlockDetails, OtsBlockTransactions, OtsContractCreator, OtsInternalOperation,
-    OtsSearchTransactions, OtsTrace,
-};
 use crate::eth::{
     error::{BlockchainError, Result},
     macros::node_info,
     EthApi,
 };
 use alloy_primitives::{Address, Bytes, B256, U256};
 use alloy_rpc_types::{
-    trace::parity::{Action, CreateAction, CreateOutput, TraceOutput},
-    Block, BlockId, BlockNumberOrTag as BlockNumber,
+    trace::{
+        otterscan::{
+            BlockDetails, ContractCreator, InternalOperation, OtsBlock, OtsBlockTransactions,
+            OtsReceipt, OtsTransactionReceipt, TraceEntry, TransactionsWithReceipts,
+        },
+        parity::{
+            Action, CallAction, CreateAction, CreateOutput, LocalizedTransactionTrace,
+            RewardAction, TraceOutput,
+        },
+    },
+    Block, BlockId, BlockNumberOrTag as BlockNumber, BlockTransactions,
 };
 use itertools::Itertools;
 
+use futures::future::join_all;
+
+pub fn mentions_address(trace: LocalizedTransactionTrace, address: Address) -> Option<B256> {
+    match (trace.trace.action, trace.trace.result) {
+        (Action::Call(CallAction { from, to, .. }), _) if from == address || to == address => {
+            trace.transaction_hash
+        }
+        (_, Some(TraceOutput::Create(CreateOutput { address: created_address, .. })))
+            if created_address == address =>
+        {
+            trace.transaction_hash
+        }
+        (Action::Create(CreateAction { from, .. }), _) if from == address => trace.transaction_hash,
+        (Action::Reward(RewardAction { author, .. }), _) if author == address => {
+            trace.transaction_hash
+        }
+        _ => None,
+    }
+}
+
+/// Converts the list of traces for a transaction into the expected Otterscan format, as
+/// specified in the [`ots_traceTransaction`](https://github.com/otterscan/otterscan/blob/develop/docs/custom-jsonrpc.md#ots_tracetransaction) spec
+pub fn batch_build_ots_traces(traces: Vec<LocalizedTransactionTrace>) -> Vec<TraceEntry> {
+    traces
+        .into_iter()
+        .filter_map(|trace| match trace.trace.action {
+            Action::Call(call) => {
+                let ots_type = serde_json::to_string(&call.call_type).unwrap();
```

Comment:
this is wrong, it writes `\"CALL\"` when it should be `"CALL"`

Reply:
@DaniPopes My bad, I didn't check that the type was serializing in lowercase and I forgot about the double quotes. 
However, `CallType` doesn't implement Display, so I'm converting it into `String` in the following way